ludwig/models/gbm.py

```diff
@@ -185,18 +220,15 @@ def load(self, save_path):
         weights_save_path = os.path.join(save_path, MODEL_WEIGHTS_FILE_NAME)
         self.lgbm_model = joblib.load(weights_save_path)
 
-        self.compile()
-
-        device = torch.device(get_torch_device())
-        self.compiled_model.to(device)
-
     def to_torchscript(self, device: Optional[TorchDevice] = None):
         """Converts the ECD model as a TorchScript model."""
-
-        # Disable gradient calculation for hummingbird Parameter nodes.
-        self.compiled_model.model.requires_grad_(False)
-
-        return super().to_torchscript(device)
+        with self.compile():
+            # Disable gradient calculation for hummingbird Parameter nodes.
+            device = torch.device(get_torch_device())
```

Comment:
I think `torch.device` is redundant with `get_torch_device()` here, we typically just do `model.to(get_torch_device())`.